### PR TITLE
Implementing relative accuracy for UncertainData

### PR DESCRIPTION
**Pull request by lawrence-hwang**
Add metric for the relative accuracy at prediction time lambda

### CODE REVIEW

**Comment by lawrence-hwang** [comment]
suggestion from Chris:
return {key: 1 - (abs(ground_truth[k] - v)/ground_truth[k]) for k,v in self.mean}

File: src/prog_algs/uncertain_data/uncertain_data.py [hunk 1]
```diff
@@ -60,6 +60,19 @@ def cov(self) -> array:
...
 
+    def relative_accuracy(self, ground_truth : dict) -> dict:
+        """The relative accuracy measure of error in RUL prediction relative to the actual RUL at a specific time index
+        
+        Returns:
+            dict(str:float): Relative accuracy for each event where value is relative accuracy between [0,1]
+        """
+        # if this check isn't here, goes to divide by zero check and raises AttributeError instead of TypeError. Keep? There are unittests checking for type
+        if not isinstance(ground_truth, dict):
+            raise TypeError("Ground truth must be passed as a dictionary argument.")
```
Comment (teubert) [inline src/prog_algs/uncertain_data/uncertain_data.py:71]: Ground truth could also be a *.container. We should probably check that out too. Maybe we should make UserDict a base for *.container, or manually check for it.

File: tests/test_uncertain_data.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Copyright © 2021 United States Government as represented by the Administrator of the National Aeronautics and Space Administration. All Rights Reserved.
 
+from typing import Type
```
Comment (teubert) [inline tests/test_uncertain_data.py:3]:
```suggestion
```
Is this needed?

**Review by teubert — CHANGES_REQUESTED** [review]

File: src/prog_algs/uncertain_data/uncertain_data.py [hunk 3]
```diff
@@ -60,6 +60,23 @@ def cov(self) -> array:
             array[array[float]]: Covariance matrix
         """
 
+    def relative_accuracy(self, ground_truth : dict) -> dict:
+        """The relative accuracy of the mean of the distribution. 
+        
+        RA = 1 - |r-p|/r 
+        Where r is ground truth RUL and p is predicted RUL distribution
```
Comment (lawrence-hwang) [inline src/prog_algs/uncertain_data/uncertain_data.py:67]: r is ground truth
p is predicted mean distribution

File: src/prog_algs/uncertain_data/uncertain_data.py [hunk 4]
```diff
@@ -60,6 +60,23 @@ def cov(self) -> array:
...
+        Prognostics (Goebel et al, 239)
```
Comment (lawrence-hwang) [inline src/prog_algs/uncertain_data/uncertain_data.py:68]: use full name of book. Prognostics: The Science of Making Predictions

File: src/prog_algs/uncertain_data/uncertain_data.py [hunk 5]
```diff
@@ -60,6 +61,23 @@ def cov(self) -> array:
...
+        
+        RA = 1 - |r-p|/r 
+        Where r is ground truth and p is mean of predicted distribution
+        Prognostics: The Science of Making Predictions (Goebel et al, 239)
+
+        Returns:
+            dict(str:float): Relative accuracy for each event where value is relative accuracy between [0,1]
+        """
+        # if this check isn't here, goes to divide by zero check and raises AttributeError instead of TypeError. Keep? There are unittests checking for type
+        if (not isinstance(ground_truth, dict)) and (not isinstance(ground_truth, DictLikeMatrixWrapper)):
```
Comment (teubert) [inline src/prog_algs/uncertain_data/uncertain_data.py:75]:
```suggestion
        if not (isinstance(ground_truth, dict) or isinstance(ground_truth, DictLikeMatrixWrapper)):
```

**Review by teubert — APPROVED** [review]

**Pull request merged**